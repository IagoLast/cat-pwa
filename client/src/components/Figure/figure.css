.fig {
  position: relative;
  display: inline-block;
  overflow: hidden;
  min-width: 200px;
  max-width: 400px;
  box-shadow: 0px 1px 1px 0px rgba(78, 78, 78, 0.6);
  padding: 12px;
  margin: 8px;
  border-radius: 2px;
  cursor: pointer;
  transition: all .3s;
  vertical-align: middle;
  background: #fff;
}

.fig__img {
  display: block;
  width: 100%;
}

.fig__caption {
  position: absolute;
  top: -100%;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 8px;
  background: rgba(255, 255, 255, .95);
  text-align: center;
  opacity: 0;
  transition: opacity .6s, top .4s;
  font-weight: 100;
}

.fig:hover .fig__caption {
  opacity: 1;
  top: 0;
}

@media (min-width: 550px) {
  .fig {
    max-width: 500px;
  }
}