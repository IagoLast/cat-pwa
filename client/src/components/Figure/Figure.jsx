import React, { Component } from 'react';
import './figure.css';

export default class Figure extends Component {
  render() {
    return <figure onClick={() => this.props.handleClick(this.props.index)} className={this.props.active ? 'fig active' : 'fig'} >
      {this.renderImg()}
      {this.renderCaption()}
    </figure>
  }

  renderImg() {
    return <img className="fig__img"
      width={this.props.data.width_s}
      height={this.props.data.height_s}
      data-src={this.props.active ? this.props.data.url_l : this.props.data.url_s}
      alt={this.props.data.title}
      title={this.props.data.title} />
  }

  renderCaption() {
    if (!this.props.active) {
      return <figcaption className="fig__caption">
        {this.props.data.title} by {this.props.data.ownername}
      </figcaption>
    }
    return <figcaption className="fig__caption">
      <h1> {this.props.data.title} </h1>
      <h3>{this.props.data.ownername}</h3>
      <p>Date: {this.props.data.datetaken}</p>
      <pre>{this.props.data.tags}</pre>
    </figcaption>
  }
}
