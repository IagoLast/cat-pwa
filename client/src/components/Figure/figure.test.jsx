import React from 'react';
import renderer from 'react-test-renderer';

import Figure from './Figure';

it('renders correctly', () => {
  const data = {
    width_s: '300px',
    height_s: '300px',
    url_s: 'fake-url',
    title: 'fake-title',
    ownername: 'fake-ownername'
  }
  const tree = renderer.create(<Figure data={data}></Figure>).toJSON();
  expect(tree).toMatchSnapshot();
});