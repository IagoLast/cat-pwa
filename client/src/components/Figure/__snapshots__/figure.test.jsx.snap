// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<figure
  className="fig"
  onClick={[Function]}
>
  <img
    alt="fake-title"
    className="fig__img"
    data-src="fake-url"
    height="300px"
    title="fake-title"
    width="300px"
  />
  <figcaption
    className="fig__caption"
  >
    <div>
      <h3
        className="title"
      >
        fake-title
      </h3>
      <p>
        by
      </p>
      <p
        className="owner"
      >
        @
        fake-ownername
      </p>
    </div>
  </figcaption>
</figure>
`;
